src/Facets/WhitelistManagerFacet.sol

```diff
@@ -0,0 +1,232 @@
+        view
+        returns (address[] memory addresses)
+    {
+        return LibAllowList.getAllowedContracts();
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function isAddressWhitelisted(
+        address _address
+    ) external view returns (bool approved) {
+        return LibAllowList.contractIsAllowed(_address);
+    }
+
+    /// @inheritdoc IWhitelistManagerFacet
+    function getApprovedFunctionSelectors()
+        external
+        view
+        returns (bytes4[] memory selectors)
+    {
+        return LibAllowList.getAllowedSelectors();
+    }
+
+    /// @dev Internal function to handle whitelist addition logic
+    function _addToWhitelist(address _contractAddress) internal {
+        if (_contractAddress == address(this)) {
+            revert CannotAuthoriseSelf();
+        }
+
+        if (LibAllowList.contractIsAllowed(_contractAddress)) return;
+
+        LibAllowList.addAllowedContract(_contractAddress);
+        emit AddressWhitelisted(_contractAddress);
+    }
+
+    /// @dev Internal function to handle whitelist removal logic
+    function _removeFromWhitelist(address _address) internal {
+        if (!LibAllowList.contractIsAllowed(_address)) return;
+
+        LibAllowList.removeAllowedContract(_address);
+        emit AddressRemoved(_address);
+    }
+
+    /// @dev Internal function to handle function selector approval logic
+    function _setFunctionApproval(bytes4 _selector, bool _approval) internal {
+        bool currentlyApproved = LibAllowList.selectorIsAllowed(_selector);
+
+        if (_approval != currentlyApproved) {
+            if (_approval) {
+                LibAllowList.addAllowedSelector(_selector);
+            } else {
+                LibAllowList.removeAllowedSelector(_selector);
+            }
+            emit FunctionSelectorApprovalChanged(_selector, _approval);
+        }
+    }
+
+    /// Temporary methods for migration ///
+    /// @dev Remove these methods after migration is complete in next facet upgrade.
+    /// @inheritdoc IWhitelistManagerFacet
+    function migrate(
+        bytes4[] calldata _selectorsToRemove,
+        address[] calldata _contractsToAdd,
+        bytes4[] calldata _selectorsToAdd
+    ) external {
+        if (msg.sender != LibDiamond.contractOwner()) {
+            LibAccess.enforceAccessControl();
+        }
+
+        LibAllowList.AllowListStorage storage als = _getAllowListStorage();
+
+        // return early if already migrated
+        if (als.migrated) return;
+
+        // clear old state
+        // reset contractAllowList
+        for (uint256 i = 0; i < als.contracts.length; i++) {
+            address contractAddr = als.contracts[i];
+            als.contractAllowList[contractAddr] = false;
+        }
+
+        // reset selectorAllowList with external selectors array because new selectors array does not exist yet
+        for (uint256 i = 0; i < _selectorsToRemove.length; i++) {
+            bytes4 selector = _selectorsToRemove[i];
```

Comment:
just a general comment (not super important here):

Storing values in a variable makes sense if the variable is used several times imho. In both loops the variable is only used once and it's newly initiated on every loop run. It's a gas-inefficient pattern.